t/24-worker-jobs.t

```diff
@@ -204,22 +204,23 @@ $engine_mock->redefine(
 my $job_mock            = Test::MockModule->new('OpenQA::Worker::Job');
 my $default_shared_hash = {upload_result => 1, uploaded_files => [], uploaded_assets => []};
 shared_hash $default_shared_hash;
-$job_mock->redefine(
-    _upload_log_file => sub {
-        my ($self, @args) = @_;
-        my $shared_hash = shared_hash;
-        push @{$shared_hash->{uploaded_files}}, \@args;
-        shared_hash $shared_hash;
-        return $shared_hash->{upload_result};
-    });
-$job_mock->redefine(
-    _upload_asset => sub {
-        my ($self, @args) = @_;
-        my $shared_hash = shared_hash;
-        push @{$shared_hash->{uploaded_assets}}, \@args;
-        shared_hash $shared_hash;
-        return $shared_hash->{upload_result};
-    });
+
+sub upload_mock {
+    # uncoverable subroutine
+    my ($key, $self, @args) = @_;     # uncoverable statement
+    my $shared_hash = shared_hash;    # uncoverable statement
+
+    # uncoverable statement count:1
+    # uncoverable statement count:2
+    push @{$shared_hash->{$key}}, \@args;
+    shared_hash $shared_hash;                # uncoverable statement
+    return $shared_hash->{upload_result};    # uncoverable statement
```

Comment:
I suppose these statements are only not tracked because this function runs within a sub process. (Otherwise the use of shared has wouldn't make any sense.) So maybe we handle this in the same way as other sub processes?

Reply:
I thought the problem is rather the mocking. But other mocked functions are still properly covered. I don't know what's wrong here

Reply:
> I don't know what's wrong here

So may explanation that this runs in a sub process which prevents the coverage tracking it wrong? Note that with "handle this in the same way as other sub processes" I meant specifically to use `Devel::Cover::report() if Devel::Cover->can('report');` here.

But of course it is questionable whether we actually want this because this is just about test code anyways.

Reply:
Well, I don't know where we actually start such subprocess. However, I tried with a `Devel::Cover::report() if Devel::Cover->can('report');` and that failed the tests reproducibly with

```
    not ok 14 - expected REST-API calls happened
    #   Failed test 'expected REST-API calls happened'
    #   at t/24-worker-jobs.t line 762.
    #     Structures begin differing at:
    #          $got->[4]{json}{status}{test_order} = Does not exist
    #     $expected->[4]{json}{status}{test_order} = ARRAY(0x560f2f589050)
```

Maybe the additional line is making some internal REST-API call timeout. I would leave it as is if you are ok with that.

Reply:
I'm ok with leaving it as it is.

Out of my head I suppose the forking happens in the upload code of the worker (which is under test here).

Reply:
ok, thx. Maybe you wanna give it a try or we can hack on this together another day. I would like that :) Wanna approve the PR?